packages/amplify-graphql-transformer-core/src/transformer-context/resolver.ts

```diff
@@ -162,14 +162,59 @@ export class TransformerResolver implements TransformerResolverProvider {
       slotEntry = [];
     }
 
-    slotEntry.push({
-      requestMappingTemplate,
-      responseMappingTemplate,
-      dataSource,
-    });
+    if (!this.updateSlotIfExists(slotName, requestMappingTemplate, responseMappingTemplate)) {
+      slotEntry.push({
+        requestMappingTemplate,
+        responseMappingTemplate,
+        dataSource,
+      });
+    }
     this.slotMap.set(slotName, slotEntry);
   };
 
+  updateSlotIfExists = (
+    slotName: string,
+    requestMappingTemplate?: MappingTemplateProvider,
+    responseMappingTemplate?: MappingTemplateProvider,
+  ): boolean => {
+    const slotEntires = this.slotMap.get(slotName)!;
```

Comment:
typo on Entires

Reply:
You also don't need the bang here, since we're null checking down below, right?

Reply:
Updated.